### PR TITLE
#168355178 Fix admin create user

### PR DESCRIPTION
**Pull request by victor-nach**
## What does this PR do?
- Allows only to create a user with only the email as the required field

## Description of Task to be completed?
- adjust models and migration files to accept null values
- adjust controllers
- create a random password generator
- adjust test and documentation

## How should this be manually tested?
- clone this branch
- Run npm run test to test the branch
- run npm run reset:db to update database
- login with an admin to receive a token
- send a post request containing the relevant user fields to POST /api/v1/auth/signup 
- review app link - https://phoenix-bn-backend-stagi-pr-35.herokuapp.com/api/v1

## What are the relevant pivotal tracker stories?
- #168355178

### CODE REVIEW

File: src/controllers/UserController.js [hunk 1]
```diff
@@ -25,14 +25,16 @@ export default class UserController {
...
+      const { email } = req.body;
+      const randomPassword = Utils.randomPassword();
+      const password = Utils.hashPassword(randomPassword);
+      const data = await UserServices.createUser({ email, password });
       const { id, isAdmin } = data;
       const token = Utils.generateToken({ id, isAdmin });
       res.set('Authorization', `Bearer ${token}`);
       return resSuccess(res, 201, data);
     } catch (error) {
+      console.log(error)
```
Comment (voke1) [inline src/controllers/UserController.js]: why did you log try/catch error here?

File: src/models/user.js [hunk 2]
```diff
@@ -1,20 +1,20 @@
...
-    email: {
-      allowNull: false,
+    lastName: {
+      allowNull: true,
       type: DataTypes.STRING,
       unique: true
     },
     password: {
-      allowNull: false,
+      allowNull: true,
```
Comment (voke1) [inline src/models/user.js]: This should remain `false` since the super admin will have to generate a password before adding the user

Comment (victor-iheanacho) [reply src/models/user.js]: thanks, would change that

File: src/controllers/UserController.js (hunk 1 above)
Comment (victor-iheanacho) [reply src/controllers/UserController.js]: would remove that as well

**Comment by voke1** [comment]
Good job @victor-nach automated test works fine now

**Review by voke1 — APPROVED** [review]

**Review by anneKay — APPROVED** [review]

**Pull request merged**